lib/bankscrap/bbva/bank.rb

```diff
@@ -130,6 +132,19 @@ def login
           'eai_password'   => @password
         }
         post(BASE_ENDPOINT + LOGIN_ENDPOINT, fields: params)
+
+        # We also need to initialize a session
+        with_headers('Content-Type' => 'application/json') do
+          post(SESSIONS_ENDPOINT, fields: {
+            consumerID: CONSUMER_ID
+          }.to_json)
+        end
+
+        # We need to extract the "tsec" header from the last response.
+        # As the Bankscrap core library doesn't expose the headers of each response
+        # we have to use Mechanize's HTTP client "current_page" method.
+        tsec = @http.current_page.response['tsec']
```

Comment:
The thing here is that the `post`, `get`, and `put` methods from [bankscrap/bank.rb](https://github.com/bankscrap/bankscrap/blob/master/lib/bankscrap/bank.rb#L58) return the body of each response. If we changed that to return a hash with something like {:headers, :body}, or just the raw response from Mechanize, then we'll need to update all the adapters.

What I did here, accessing the `@http ` instance variable directly, that gets defined in the parent class, is definitely not clean but seemed like a good solution for the moment. 

As we have plans to do evolve the data model for the core library and that will require all adapters to be updated, I'd hold this change until then.